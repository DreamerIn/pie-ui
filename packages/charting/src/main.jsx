import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Collapsible } from '@pie-lib/render-ui';
import { Chart, chartTypes } from '@pie-lib/charting';
import isEqual from 'lodash/isEqual';


export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    onAnswersChange: PropTypes.func,
  };

  static defaultProps = { classes: {} };

  constructor(props) {
    super(props);
    const { model: { data = [] } } = props;

    this.state = { data };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { model: { data: nextData = [] } = {} } = nextProps;
    const { model: { data = [] } } = this.props;

    if (!isEqual(nextData, data)) {
      this.setState({ data: nextData })
    }
  }

  changeData = data => {
    this.setState({ data }, () => this.props.onAnswersChange(data));
  };

  render() {
    const { model, classes } = this.props;
    const { data } = this.state;
    const {
      teacherInstructions,
      prompt,
      chartType,
      size,
      domain,
      range,
      title,
      editCategoryEnabled,
      addCategoryEnabled,
      categoryDefaultLabel,
      rationale
    } = model;

    return (
      <div>
        {
          teacherInstructions && (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
            >
              <div dangerouslySetInnerHTML={{ __html: teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        <div
          className={classes.prompt}
          dangerouslySetInnerHTML={{ __html: prompt }}
        />
        <br />

        <Chart
          chartType={chartType}
          size={size}
          domain={domain}
          range={range}
          charts={[
            chartTypes.Bar(),
            chartTypes.Histogram(),
            chartTypes.LineDot(),
            chartTypes.LineCross(),
            chartTypes.DotPlot(),
            chartTypes.LinePlot()
          ]}
          data={data}
          title={title}
          onDataChange={this.changeData}
          editCategoryEnabled={editCategoryEnabled}
          addCategoryEnabled={addCategoryEnabled}
          categoryDefaultLabel={categoryDefaultLabel}
        />

        <br />
        {
          rationale && (
            <Collapsible labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}>
              <div dangerouslySetInnerHTML={{ __html: rationale }}/>
            </Collapsible>
          )
        }
      </div>
    );
  }
}

const styles = () => ({
  prompt: {
    verticalAlign: 'middle'
  }
});

export default withStyles(styles)(Main);
