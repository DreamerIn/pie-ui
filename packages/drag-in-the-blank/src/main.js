import React from 'react';
import PropTypes from 'prop-types';
import { DragInTheBlank } from '@pie-lib/mask-markup';
import { withDragContext } from '@pie-lib/drag';

const DraggableDragInTheBlank = withDragContext(DragInTheBlank);

export class Main extends React.Component {
  static propTypes = {
    prompt: PropTypes.string,
    disabled: PropTypes.bool,
    markup: PropTypes.string,
    value: PropTypes.object,
    feedback: PropTypes.object,
    onChange: PropTypes.func
  };

  static defaultProps = {
    value: {}
  };

  render() {
    const { prompt } = this.props;

    return (
      <div>
        {prompt && <div dangerouslySetInnerHTML={{ __html: prompt }}/>}
        <DraggableDragInTheBlank {...this.props} />
      </div>
    );
  }
}

export default Main;
