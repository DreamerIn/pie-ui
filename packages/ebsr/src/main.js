import MultipleChoice from '@pie-ui/multiple-choice';

export default class Main extends HTMLElement {
  static defineMultipleChoice() {
    if(!customElements.get('multiple-choice')){
      customElements.define('multiple-choice', MultipleChoice);
    }
  }

  static getParts = () => ({
    partA: document.getElementById('part-a'),
    partB: document.getElementById('part-b'),
  });

  constructor() {
    super();

    Main.defineMultipleChoice();
  }


  set modelAndSession(modelAndSession) {
    this._main = modelAndSession;

    this.updateModels();
    this.captureSessionChanges();
  }

  updateModels() {
    const { partA, partB } = Main.getParts();

    if (partA && partB) {
      this.updatePartModel(partA, 'partA');
      this.updatePartModel(partB, 'partB');
    }
  }

  updatePartModel(part, key) {
    const { model, model: { mode }, session: { value } } = this._main;
    const { correctResponse, disabled, keyMode, showCorrect, choiceMode, choices, prompt } = model[key];

    part.model = {
      id: key,
      choiceMode,
      choices,
      correctResponse,
      disabled,
      keyMode,
      mode,
      prompt,
      showCorrect
    };

    part.session = { id: key };

    if (value) {
      part.session = value[key];
    }
  }

  captureSessionChanges() {
    const { partA, partB } = Main.getParts();

    if (partA && partB) {
      this.capturePartSessionChanged(partA, 'partA');
      this.capturePartSessionChanged(partB, 'partB');
    }
  }

  capturePartSessionChanged(part, key) {
    const self = this;

    part.addEventListener('session-changed', (event)  => {
      event.stopImmediatePropagation();
      self.updateValue(event.srcElement._session, key);
    });
  }

  updateValue(partSession, key) {
    const { session } = this._main;

    const sessionRepacked = {
      ...session,
      [key]: partSession,
    };

    this._main.session = sessionRepacked;

    const event = new CustomEvent('main-session-changed', {
      detail: {
        session: sessionRepacked,
      }
    });

    this.dispatchEvent(event);
  }

  connectedCallback() {
    this._render();
  }

  _render() {
    this.innerHTML = `
      <div>
        <multiple-choice id="part-a"></multiple-choice>
        <multiple-choice id="part-b"></multiple-choice>
      </div>
    `;
  }
}
