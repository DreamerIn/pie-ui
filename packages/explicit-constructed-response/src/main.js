import React from 'react';
import PropTypes from 'prop-types';
import { ConstructedResponse } from '@pie-lib/mask-markup';

export class Main extends React.Component {
  static propTypes = {
    prompt: PropTypes.string,
    disabled: PropTypes.bool,
    markup: PropTypes.string,
    value: PropTypes.object,
    feedback: PropTypes.object,
    onChange: PropTypes.func
  };

  static defaultProps = {
    value: {}
  };

  render() {
    const { prompt } = this.props;

    return (
      <div>
        {prompt && <div dangerouslySetInnerHTML={{ __html: prompt }}/>}
        <ConstructedResponse {...this.props} />
      </div>
    );
  }
}

export default Main;
