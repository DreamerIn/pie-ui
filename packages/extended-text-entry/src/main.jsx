import React from 'react';
import EditableHTML from '@pie-lib/editable-html';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import debug from 'debug';
import { Feedback, Collapsible } from '@pie-lib/render-ui';

const log = debug('@pie-ui:extended-text-entry');

const style = theme => ({
  main: {
    margin: '20px'
  },
  prompt: {
    width: '100%',
    marginBottom: theme.spacing.unit * 2
  }
});

class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object,
    classes: PropTypes.object.isRequired,
    session: PropTypes.shape({
      value: PropTypes.string
    }).isRequired
  };

  render() {
    const { model, onChange, classes, session } = this.props;
    const { dimensions, disabled, feedback, teacherInstructions, mathInput } = model;
    const { value } = session;
    const { width, height } = dimensions || {};
    log('[render] disabled? ', disabled);

    return (
      <div className={classes.main}>
        {
          teacherInstructions && (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
              className={classes.collapsible}
            >
              <div dangerouslySetInnerHTML={{ __html: teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        {model.prompt && (
          <Typography
            className={classes.prompt}
            dangerouslySetInnerHTML={{ __html: model.prompt }}
          />
        )}
        <EditableHTML
          onChange={onChange}
          markup={value || ''}
          width={width && width.toString()}
          height={height && height.toString()}
          disabled={disabled}
          highlightShape={true}
          pluginProps={{
            math: {
              disabled: !mathInput,
              keypadMode: this.props.model.equationEditor

            }
          }}
        />
        <br />
        {feedback && <Feedback correctness="correct" feedback={feedback} />}
      </div>
    );
  }
}

export default withStyles(style)(Main);
