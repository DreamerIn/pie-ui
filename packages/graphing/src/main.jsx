import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    session: PropTypes.shape({}),
    onSessionChange: PropTypes.func,
    model: PropTypes.object.isRequired
  };

  render() {
    return (
      <div />
    );
  }
}

const styles = theme => ({});

export default withStyles(styles)(Main);