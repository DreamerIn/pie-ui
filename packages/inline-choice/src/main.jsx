import { withStyles } from 'material-ui/styles';

import InlineChoice from './inline-choice';
import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const styleSheet = () => {
  return {};
};

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onChoiceChanged: PropTypes.func,
    classes: PropTypes.object.isRequired
  };

  static defaultProps = {
    model: {},
    session: {}
  };

  constructor(props) {
    super(props);
  }

  render() {
    const { model, onChoiceChanged, session, classes } = this.props;

    return (
      <div className={classNames(classes.root, classes[model.className])}>
        {model.prompt && (
          <div dangerouslySetInnerHTML={{ __html: model.prompt }} />
        )}
        <InlineChoice
          {...model}
          session={session}
          onChoiceChanged={onChoiceChanged}
        />
      </div>
    );
  }
}

export default withStyles(styleSheet)(Main);
