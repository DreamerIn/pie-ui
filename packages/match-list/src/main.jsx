import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withDragContext, swap } from '@pie-lib/drag';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { Feedback } from '@pie-lib/render-ui';
import { withStyles } from '@material-ui/core/styles';
import uniqueId from 'lodash/uniqueId';
import isUndefined from 'lodash/isUndefined';
import findKey from 'lodash/findKey';
import Arrow from './arrow';
import AnswerList from './answer-list';
import ChoicesList from './choices-list';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    session: PropTypes.object.isRequired,
    onSessionChange: PropTypes.func,
    model: PropTypes.object.isRequired,
    prompt: PropTypes.string
  };

  constructor(props) {
    super(props);

    this.instanceId = uniqueId();
    this.state = {
      showCorrectAnswer: false
    };
  }

  onRemoveAnswer(id) {
    const { session, onSessionChange } = this.props;

    session.value[id] = undefined;

    onSessionChange(session);
  }

  onPlaceAnswer(place, id) {
    const { session, onSessionChange, model } = this.props;
    const { config: { duplicates } } = model;

    if (isUndefined(session.value)) {
      session.value = {};
    }

    const choiceKey = findKey(session.value, val => val === id);

    if (choiceKey && !duplicates) {
      session.value = swap(session.value, choiceKey, place);
    } else {
      session.value[place] = id;
    }

    onSessionChange(session);
  }

  toggleShowCorrect = () => {
    this.setState({ showCorrectAnswer: !this.state.showCorrectAnswer });
  };

  render() {
    const { showCorrectAnswer } = this.state;
    const { classes, model, session } = this.props;
    const { config, mode } = model;
    const { prompt } = config;

    return (
      <div className={classes.mainContainer}>
        <CorrectAnswerToggle
          show={mode === 'evaluate'}
          toggled={showCorrectAnswer}
          onToggle={this.toggleShowCorrect}
        />
        <div
          className={classes.prompt}
          dangerouslySetInnerHTML={{ __html: prompt }}
        />
        <div className={classes.listContainer}>
          <div className={classnames(classes.itemList, classes.promptList)}>
            {
              config.prompts.map((pr) => (
                <div
                  key={pr.id}
                  className={classes.promptEntry}
                  dangerouslySetInnerHTML={{ __html: pr.title }}
                />
              ))
            }
          </div>
          <div className={classnames(classes.itemList, classes.arrowList)}>
            {
              config.prompts.map((pr) => (
                <div
                  key={pr.id}
                  className={classes.arrowEntry}
                >
                  <Arrow direction="left" />
                  <Arrow />
                </div>
              ))
            }
          </div>
          <AnswerList
            instanceId={this.instanceId}
            model={model}
            session={session}
            onPlaceAnswer={(place, id) => this.onPlaceAnswer(place, id)}
            onRemoveAnswer={id => this.onRemoveAnswer(id)}
            disabled={mode !== 'gather'}
            showCorrect={showCorrectAnswer}
          />
        </div>
        <ChoicesList
          instanceId={this.instanceId}
          model={model}
          session={session}
          disabled={mode !== 'gather'}
        />
        {
          model.correctness &&
          model.feedback &&
          !showCorrectAnswer &&(
            <Feedback
              correctness={model.correctness.correctness}
              feedback={model.feedback}
            />
          )
        }
      </div>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center'
  },
  listContainer: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'space-between'
  },
  itemList: {
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  arrowList: {
    flex: 1,
    width: '100%'
  },
  promptList: {
    alignItems: 'flex-start'
  },
  main: {
    width: '100%'
  },
  toggle: {
    paddingBottom: theme.spacing.unit * 3
  },
  prompt: {
    verticalAlign: 'middle'
  },
  promptEntry: {
    border: '1px solid #c2c2c2',
    boxSizing: 'border-box',
    height: 40,
    overflow: 'hidden',
    margin: '10px 0',
    width: '100%',
    textAlign: 'center',
    padding: 10
  },
  arrowEntry: {
    alignItems: 'normal',
    display: 'flex',
    height: 40,
    margin: '10px 20px',
    width: '100%'
  }
});

export const styledMain = withStyles(styles)(Main);

export default withDragContext(styledMain);