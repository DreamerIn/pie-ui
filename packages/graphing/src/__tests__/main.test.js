import * as React from 'react';
import { Main } from '../main';
import { shallow } from 'enzyme/build';

describe('Main', () => {
  const defaultProps = {
    model: {
      backgroundMarks: [],
      correctMarks: [],
    },
    onSessionChange: jest.fn(),
    onAnswersChange: jest.fn(),
    session: {}
  };

  describe('render', () => {
    let w;

    beforeEach(() => {
      w = props => shallow(<Main { ...props } />);
    });

    it('snapshot', () => {
      expect(w(defaultProps)).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let w;

    beforeEach(() => {
      w = props => shallow(<Main { ...props } />);
    });

    it('calls onAnswersChange', () => {
      let wrapper = w(defaultProps);
      wrapper.instance().changeMarks([]);

      expect(wrapper.instance().props.onAnswersChange).toHaveBeenCalledWith([]);
      expect(wrapper.instance().state.model.marks).toEqual([]);
    });
  });

});
