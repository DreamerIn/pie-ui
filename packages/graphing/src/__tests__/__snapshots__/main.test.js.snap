// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render snapshot 1`] = `
<div>
  <Root
    backgroundMarks={Array []}
    marks={Array []}
    onChangeMarks={[Function]}
    size={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;
